linux_sql/scripts/psql_docker.sh

```diff
@@ -1,47 +1,49 @@
 #!/bin/bash
 #Setup arguments
+script_execute=$1
+PGUSER=$2
+PGPASSWORD=$3
 
 #Main code
-#check if docker_deamon is running - otherwise start run
+#check if docker_deamon is running - otherwise start
 systemctl status docker || systemctl start docker
 
 #check if user wants to create new psql table
-if [ $1 == "create" ]; then
+if [ $script_execute == "create" ]; then
   #list all docker containers and filter for jrvs-psql and then counts words in name
   if  [ "$(docker ps -a -f name=jrvs-psql | wc -l)" == "2" ]; then
     echo "Error: Container already exists"
     exit 1
 
-  elif [ "$#" != 4 ]; then
+  elif [ "$#" != 3 ]; then
     echo "Error: database username or password not passed"
     exit 1
 
   else
     docker pull postgres
     docker volume create pgdata
-    export db_user=$2
-    export db_pass=$3
-    docker run --name jrvs-psql -e POSTGRES_PASSWORD=$db_pass \
-    -e POSTGRES_USER=$db_user -d -v pgdata:/var/lib/postgresql/data -p 5432:5432 postgres
+
+    docker run --name jrvs-psql -e POSTGRES_PASSWORD=$PGPASSWORD \
+    -e POSTGRES_USER=$PGUSER -d -v pgdata:/var/lib/postgresql/data -p 5432:5432 postgres
     exit $?
     fi
 
-elif  [ "$(docker ps -a -f name=jrvs-psql | wc -l)" != "2" ]; then
-    echo "Error: Container not created"
+else
+  if [ "$(docker ps -a -f name=jrvs-psql | wc -l)" != "2" ]; then
+    echo "Error: Container does not exist"
     exit 1
+    fi
 
-else
-  if [ $1 == "start" ]; then
+  if [ $script_execute == "start" ]; then
     docker container start jrvs-psql
     exit $?
```

Comment:
what happens if container was not created?

Reply:
it is checked by if statement in line 32